azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.execapp;
+
+import azkaban.executor.ExecutableFlow;
+import azkaban.project.ProjectFileHandler;
+import azkaban.project.ProjectLoader;
+import azkaban.project.ProjectManagerException;
+import azkaban.utils.FileIOUtils;
+import azkaban.utils.Pair;
+import azkaban.utils.Utils;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+
+import static com.google.common.base.Preconditions.*;
+import static java.util.Objects.*;
+
+
+public class FlowPreparer {
+  private static final Logger log = Logger.getLogger(FlowPreparer.class);
+
+  // TODO move to config class
+  private final File executionsDir;
+  // TODO move to config class
+  private final File projectsDir;
+
+  private final Map<Pair<Integer, Integer>, ProjectVersion> installedProjects;
+  private final ProjectLoader projectLoader;
+
+  public FlowPreparer(ProjectLoader projectLoader,
+      File executionsDir,
+      File projectsDir,
+      Map<Pair<Integer, Integer>, ProjectVersion> installedProjects) {
+    this.projectLoader = projectLoader;
+    this.executionsDir = executionsDir;
+    this.projectsDir = projectsDir;
+    this.installedProjects = installedProjects;
+  }
+
+  /**
+   * Prepare the flow directory for execution.
+   *
+   * @param flow Executable Flow instance.
+   */
+  void setup(ExecutableFlow flow) {
```

Comment:
It currently doesn't require any public access since it is used just by the executor to prepare a working directory.